syft/pkg/cataloger/dotnet/licenses.go

```diff
@@ -0,0 +1,608 @@
+			if err != nil {
+				continue
+			}
+
+			assets = append(assets, *assetDefinition)
+		}
+
+		if len(assets) == 0 {
+			err = fmt.Errorf("could not retrieve any asset definitions")
+		}
+	}
+
+	return assets, err
+}
+
+func extractProjectAssetsFromResolvedFile(resolver file.Resolver, l file.Location) (asset *projectAssets, err error) {
+	contentReader, err := resolver.FileContentsByLocation(l)
+	if err != nil {
+		return nil, err
+	}
+	defer internal.CloseAndLogError(contentReader, l.RealPath)
+
+	assetFileData, err := io.ReadAll(contentReader)
+	if err != nil {
+		return nil, err
+	}
+
+	asset = &projectAssets{}
+	if err = json.Unmarshal(assetFileData, asset); err != nil {
+		return nil, err
+	}
+
+	return asset, nil
+}
+
+func getNuGetCachesFromProjectAssets(assets []projectAssets) []string {
+	paths := []string{}
+
+	// Try to determine NuGet package folders from project assets
+	if len(assets) > 0 {
+		for _, assetDefinition := range assets {
+			for folder := range assetDefinition.PackageFolders {
+				found := false
+				for _, known := range paths {
+					if known == folder {
+						found = true
+						break
+					}
+				}
+				if !found {
+					paths = append(paths, folder)
+				}
+			}
+		}
+	}
+
+	return paths
+}
+
+func parseSDKLocalCachePathsOutput(outputLines []string) []string {
+	paths := []string{}
+
+	for _, line := range outputLines {
+		line = strings.TrimSpace(line)
+		if lineParts := strings.Split(line, ": "); len(lineParts) == 2 {
+			folder := lineParts[1]
+			found := false
+			for _, known := range paths {
+				if known == folder {
+					found = true
+					break
+				}
+			}
+			if !found {
+				paths = append(paths, folder)
+			}
+		}
+	}
+
+	return paths
+}
+
+func getNuGetCachesFromSDK() []string {
+	paths := []string{}
+
+	// Query NuGet itself for its cache locations
+	if dotnetPath, err := determineDotnetExecutablePath(); err == nil {
+		ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+		defer cancel()
+
+		// cf. https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-nuget-locals
+		cmd := exec.CommandContext(ctx, dotnetPath, "nuget", "locals", "all", "-l", "--force-english-output")
+		if stdout, err := cmd.StdoutPipe(); err == nil {
+			if err := cmd.Start(); err == nil {
+				if data, err := io.ReadAll(stdout); err == nil {
+					lines := strings.Split(string(data), "\n")
+					paths = parseSDKLocalCachePathsOutput(lines)
+				}
+			}
+		}
+	}
+
+	return paths
+}
+
+func (c *nugetLicenseResolver) getLocalNugetFolderResolvers(assetDefinitions []projectAssets) []file.Resolver {
+	nugetPackagePaths := []string{}
+	if len(c.opts.LocalCachePaths) > 0 {
+		nugetPackagePaths = append(nugetPackagePaths, c.opts.LocalCachePaths...)
+	} else {
+		nugetPackagePaths = append(nugetPackagePaths, getNuGetCachesFromProjectAssets(assetDefinitions)...)
+
+		nugetPackagePaths = append(nugetPackagePaths, getNuGetCachesFromSDK()...)
+	}
```

Comment:
Not knowing much about NuGet -- are these mutually exclusive? should we just always append all the paths we can find?